### PR TITLE
V3—dot_plot_with_categorical

### PR DESCRIPTION
**Pull request by bfinzer**
With a numeric attribute on one axis and categorical on the other, the plot is a dot plot split by categories.

The base lines of each of the distributions are not yet drawn.

### CODE REVIEW

File: v3/src/components/graph/models/axis-model.ts [hunk 1]
```diff
@@ -12,9 +12,27 @@ export const attrPlaceToAxisPlace: Partial<Record<GraphAttrPlace, AxisPlace>> =
...
+      attrPlace = 'y'
+      break
+    case 'top':
+      attrPlace = 'topSplit'
+      break
+    case 'right':
+      attrPlace = 'y2'  // Todo: how to deal with 'rightSplit'?
+  }
+  return attrPlace
+}
```
Comment (kswenson) [inline v3/src/components/graph/models/axis-model.ts:31]: Seems like you could/should follow the same pattern as `attrPlaceToAxisPlace` for consistency (and brevity):
```suggestion
export const axisPlaceToAttrPlace: Record<AxisPlace, GraphAttrPlace> = {
  bottom: "x",
  left: "y",
  top: "topSplit",
  right: "y2"  // Todo: how to deal with 'rightSplit'?
}
```

File: v3/src/components/graph/models/data-configuration-model.ts [hunk 2]
```diff
@@ -101,6 +101,11 @@ export const DataConfigurationModel = types
       else {
         self.attributeDescriptions.delete(place)
       }
+    },
+    setPrimaryPlace( aPlace: GraphAttrPlace) {
+      if( aPlace === 'x' || aPlace === 'y') {
```
[the remark this answers is not in the document]
Comment (bfinzer) [reply v3/src/components/graph/models/data-configuration-model.ts:106]: Added test that calls setPrimaryPlace

File: v3/src/components/graph/models/graph-controller.ts [hunk 3]
```diff
@@ -55,38 +56,46 @@ export class GraphController {
...
       currentAxisType = axisModel?.type,
       plotChoices: Record<string, Record<string, PlotType>> = {
         empty: {empty: 'casePlot', numeric: 'dotPlot', categorical: 'dotChart'},
         numeric: {empty: 'dotPlot', numeric: 'scatterPlot', categorical: 'dotPlot'},
         categorical: {empty: 'dotChart', numeric: 'dotPlot', categorical: 'dotChart'}
       },
-      attrIDs: string[] = []
-    attributeType !== 'empty' && attrIDs.push(attrID)
-    otherAttributeType !== 'empty' && attrIDs.push(otherAttrID)
+      graphAttributePlace = axisPlaceToGraphAttrPlace(axisPlace),
```
[the remark this answers is not in the document]
Comment (bfinzer) [reply v3/src/components/graph/models/graph-controller.ts:86]: Good catch.

**Review by kswenson — APPROVED** [review]
👍 Changes look good -- just one missing `useEffect` dependency (apparently).

**Pull request merged**